File: src/Creational_Patterns/Prototype/examples/go/prototype.go
package main

import "fmt"

type Car struct {
    Brand string
    Model string
    Color string
}

func (c *Car) Clone() *Car {
    return &Car{
        Brand: c.Brand,
        Model: c.Model,
        Color: c.Color,
    }
}

func main() {
    carPrototype := &Car{
        Brand: "Toyota",
        Model: "Corolla",
        Color: "Blue",
    }

    car1 := carPrototype.Clone()
    car2 := carPrototype.Clone()

    car1.Color = "Red"

    fmt.Println("Car 1:", car1.Brand, car1.Model, car1.Color)
    fmt.Println("Car 2:", car2.Brand, car2.Model, car2.Color)
}

File: src/Creational_Patterns/Prototype/examples/rust/prototype.rs
#[derive(Clone)]
struct Car {
    brand: String,
    model: String,
    color: String,
}

impl Car {
    fn new(brand: &str, model: &str, color: &str) -> Self {
        Car {
            brand: String::from(brand),
            model: String::from(model),
            color: String::from(color),
        }
    }

    fn print_info(&self) {
        println!("Brand: {}, Model: {}, Color: {}", self.brand, self.model, self.color);
    }
}

fn main() {
    let car_prototype = Car::new("Toyota", "Corolla", "Blue");
    let car1 = car_prototype.clone();
    let car2 = car_prototype.clone();

    car1.print_info();
    car2.print_info();
}